### ISSUE TITLE
Allow customizing Dokka options

### ISSUE THREAD

**Issue by tadfisher** [JavaEden/Orchid#222]
Right now, the repositories used for dependency resolution by `dokka-json` are fixed, so it doesn't look like we can, for example, add https://maven.google.com/ to resolve Android artifacts.

It would also be nice to be able to specify other Dokka options, such as the source roots to include or the Kotlin tasks to analyze.

**Comment by cjbrooks12** [comment]
The `KotlinDoc` plugin isn't connected to your Gradle build, it is intended to be run completely independent of Gradle or any other build tool. Unfortunately, this does mean that it is limited in how it runs, not having knowledge of Gradle's sourcesets, dependencies, and tasks, but also means there's really no need to include other package repositories in the Dokka run. 

But you are able to [customize the source roots](https://orchid.netlify.com/plugins/OrchidKotlindoc/docs) to analyze, with the `kotlindoc. sourceDirs` property in your `config.yml`, which accepts a list of file paths relative to your Orchid site resources dir:

```yml
# config.yml
...
kotlindoc:
  sourceDirs: 
    - '../../main/kotlin'
```

That being said, Dokka does have some issues when referencing classes not in the stdlib or in the sources being analyzed. The intention is that the docs will still work when referencing these classes and they just won't be linkable, but they are currently showing up empty. I will have to investigate this further.

**Comment by tadfisher** [comment]
> Unfortunately, this does mean that it is limited in how it runs, not having knowledge of Gradle's sourcesets, dependencies, and tasks, but also means there's really no need to include other package repositories in the Dokka run.

Understandable, but perhaps it would be possible to supply a classpath to the plugin, and the Gradle integration could infer these values from the project?

> The intention is that the docs will still work when referencing these classes and they just won't be linkable, but they are currently showing up empty.

This is because Dokka takes an additional "classpath" parameter that is used for resolving types in dependency jars: https://github.com/Kotlin/dokka#using-the-command-line

I don't believe the Kotlindoc plugin is supplying this parameter, thus Dokka outputs `<ERROR CLASS>` (at least in its Markdown/HTML output) for the unknown types.

**Comment by cjbrooks12** [comment]
I have the plugin passing the classpath through to Dokka. I am releasing it now, it will be available in 0.15.3 later today. There is a new `--kotlindocClasspath` arg that you can pass to Orchid, which gets passed-through directly to Dokka's `-classpath` arg. 

Here's how it would look in Gradle:

```groovy
afterEvaluate { // I was getting Gradle errors unless the `orchid` closure was in `afterEvaluate`
    orchid {
        ...
        args = [
            "--kotlindocClasspath", // the new Orchid arg
            project(":app").sourceSets.main.runtimeClasspath.getAsPath() // use Gradle APIs to get the classpath to pass-through
        ]
    }
}
```

**Comment by cjbrooks12** [comment]
0.15.3 is now available, please try it out and let me know how it works! Here are the [updated docs](https://orchid.netlify.com/plugins/OrchidKotlindoc)
